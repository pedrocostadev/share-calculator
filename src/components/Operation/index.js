import { connect } from 'react-redux';

import Operation from './operation';
import { operationsSelectors } from '../../selectors';

const mapStateToProps = state => ({
  text: operationsSelectors.selectOperation(state),
});

export default connect(mapStateToProps)(Operation);
