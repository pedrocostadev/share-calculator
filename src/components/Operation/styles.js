import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  operation: {
    fontSize: 28,
    color: '#FFFFFF',
    padding: 10,
    textAlign: 'right',
  },
});

export default styles;
