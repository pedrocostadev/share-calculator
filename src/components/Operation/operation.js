import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import styles from './styles';

const Operation = ({ text }) => <Text style={styles.operation}>{text}</Text>;

Operation.propTypes = {
  text: PropTypes.string,
};

Operation.defaultProps = {
  text: '',
};

export default Operation;
