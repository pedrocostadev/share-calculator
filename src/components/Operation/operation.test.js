import React from 'react';
import Operation from './operation';
import { render, cleanup } from '@testing-library/react-native';

describe('<Operation />', () => {
  afterEach(cleanup);

  test('should render', async () => {
    const text = '12 + 12';

    const { queryAllByText, container } = render(<Operation text={text} />);
    const ocurrences = await queryAllByText(text);

    expect(ocurrences).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
