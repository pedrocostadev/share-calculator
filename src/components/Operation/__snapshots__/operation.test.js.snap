// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Operation /> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 28,
        "padding": 10,
        "textAlign": "right",
      }
    }
  >
    12 + 12
  </Text>
</View>
`;
